core/common/src/main/java/alluxio/underfs/Fingerprint.java

```diff
@@ -33,6 +33,11 @@
   /** These tags are optional, and are serialized after the required tags. */
   private static final Tag[] OPTIONAL_TAGS = {Tag.CONTENT_HASH};
 
+  /** These tags are all the metadata tags in the fingerprints. */
+  private static final Tag[] METADATA_TAGS = {Tag.TYPE, Tag.UFS, Tag.OWNER, Tag.GROUP, Tag.MODE};
```

Comment:
`TYPE` should also be a content tag - either going from no content to some or vice versa. Similar for `UFS` where if it changes we would want to invalidate both content and metadata